lib/tasks/database_updates/backfill_publisher_selected_wallet.rake

```diff
@@ -0,0 +1,105 @@
+namespace :database_updates do
+  desc 'Backfill Publisher Selected Wallets'
+  task :backfill_publisher_selected_wallets => :environment do
+    def handle_gemini
+      now = Time.zone.now
+      gemini_to_set = Publisher.
+        includes([:gemini_connection]).
+        joins(:gemini_connection).
+        where('gemini_connections.is_verified = TRUE').
+        where('publishers.email IS NOT NULL').
+        where('publishers.selected_wallet_provider_id IS NULL').
+        pluck(:id, 'gemini_connections.id') # use pluck to avoid loading user_authentication_token
+
+      # For testing in dev
+      # publisher_ids = GeminiConnection.first(3).map { |p| p.publisher.id }
+      # gemini_to_set = Publisher.where(id: publisher_ids).
+      #   includes([:gemini_connection]).
+      #   joins(:gemini_connection).
+      #   pluck(:id, 'gemini_connections.id')
+
+      gemini_new_records = gemini_to_set.map do |publisher_gemini|
+        {
+          id: publisher_gemini[0], # publisher id
+          selected_wallet_provider_id: publisher_gemini[1], # gemini connections id
+          selected_wallet_provider_type: 'GeminiConnection',
+          created_at: now,
```

Comment:
We should not be updating `created_at`

Reply:
One of the current [downsides of upsert_all](https://medium.com/@retrorubies/upcoming-rails-6-bulk-insert-upsert-feature-2d642419557d). We could change the defaults [in a migration](https://stackoverflow.com/a/65546384), I'll work on a migration tomorrow.

Reply:
I really am against updating the `created_at` field because we use it for metrics like when user's joined. Alternatively you could just refactor this to use https://stackoverflow.com/questions/6629088/bulk-record-update-with-sql

Also totally fine with an N+1 query given that there's 448 uphold accounts that are missing a type.

Reply:
This has been fixed with the migration.